File: codewars/6kyu/split-strings/dduck-split-strings.py
# split-strings
# https://www.codewars.com/kata/515de9ae9dcfc28eb6000001

from unittest import TestCase


def solution(s):
    string = s + "_" if len(s) % 2 == 1 else s
    return [string[i:i + 2] for i in range(0, len(string), 2)]


tests = (
    ("asdfadsf", ['as', 'df', 'ad', 'sf']),
    ("asdfads", ['as', 'df', 'ad', 's_']),
    ("", []),
    ("x", ["x_"]),
)

for inp, exp in tests:
    TestCase().assertEqual(solution(inp), exp)

File: codewars/6kyu/split-strings/dduck-split-strings.js
// split-strings
// https://www.codewars.com/kata/515de9ae9dcfc28eb6000001

const solution = (str) => {
  if (str.length % 2 === 1) {
    str += '_';
  }
  
  let result = [];
  
  while (str.length) {
    result.push(str.slice(0, 2));
    str = str.slice(2);
  }
  
  return result;
};

console.log(solution("abcdef"), ["ab", "cd", "ef"]);
console.log(solution("abcdefg"), ["ab", "cd", "ef", "g_"]);
console.log(solution(""), []);
